### PR TITLE
Island chat spy

### PR DESCRIPTION
**Pull request by nik2143**

### CODE REVIEW

**Comment by IridiumLLC** [comment]
`public List<String> spyingIslandsChat = new ArrayList<>();`
Use OOP for this, add a boolean value inside the usersclass for island chat then do User.getUser(Player) to access the class

**Comment by IridiumLLC** [comment]
Although that would make it harder to loop :thinking:

**Comment by nik2143** [comment]
> `public List<String> spyingIslandsChat = new ArrayList<>();`
> Use OOP for this, add a boolean value inside the usersclass for island chat then do User.getUser(Player) to access the class
I have thought to do it but I haven't found a way to loop excluding loop for all online users and checking if they have the value

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerJoinLeaveListener.java [hunk 1]
```diff
@@ -49,4 +52,14 @@ public void onJoin(PlayerJoinEvent event) {
...
     }
+
+    @EventHandler
+    public void onLeave(PlayerQuitEvent event){
+        try {
+            IridiumSkyblock.getInstance().spyingIslandsChat.remove(event.getPlayer());
+        }
+        catch (Exception e) {
+            IridiumSkyblock.getInstance().sendErrorMessage(e);
+        }
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerJoinLeaveListener.java:63]: This code won't throw an exception. Even if it did, the plugin would blow up first and then this exception is probably the smallest problem.

**Review by dlsf — CHANGES_REQUESTED** [review]

**Comment by dlsf** [comment]
> I have thought to do it but I haven't found a way to loop, excluding loop for all online users and checking if they have the value
I don't see a problem with this solution. The required methods are so fast that this should not cause a performance issue.

**Comment by IridiumLLC** [comment]
Could just use streams to make things easier:
`Bukkit.getServer().getOnlinePlayers().stream().filter(player -> User.getUser(player).spying).collect(Collectors.toList());`

**Pull request merged**